rest/admin_api.go

```diff
@@ -365,7 +384,11 @@ func (h *handler) handlePutDbConfig() (err error) {
 					return nil, err
 				}
 
-				// TODO: CBG-1452 Optimistic Concurrency Control/RevID check
+				headerVersion := h.rq.Header.Get("If-Match")
+				if headerVersion != "" && headerVersion != bucketDbConfig.Version {
+					return nil, base.HTTPErrorf(http.StatusConflict, "Provided If-Match header does not match current config version")
```

Comment:
`http.StatusConflict` is used throughout SG for the same type of situation on docs, but RFC7232 suggests that `http.StatusPreconditionFailed` is the appropriate status code when a precondition header like `If-Match` is not satisfied.